app/controllers/tag_controller.rb

```diff
@@ -73,6 +73,15 @@ def index
   end
 
   def show
+
+    # Enhancement #6306 - Add counts to `by type` dropdown on tag pages 
+    @counts = {:posts => 0, :questions => 0, :wiki => 0 }
+    @counts[:posts] = Tag.find_nodes_by_type([params[:id]], 'note', false).count
+    @counts[:questions] = Tag.find_nodes_by_type("question:#{params[:id]}", 'note', false).count
+    @counts[:wiki] = Tag.find_nodes_by_type([params[:id]], 'page', false).count
+    params[:counts] = @counts
```

Comment:
Actually I realized it's possible to just directly use `@counts` as `@counts[:wiki]` in the templates, so perhaps there could be a slight refactoring to simplify that? It's standard practice to use an instance variable (w @) to pass values from the controller to the views. Not that it's wrong here! But if you'd like to make that modification in a new PR that'd be awesome!

Thanks again!!!

Reply:
I got it! you are right! I will refactor that ;)